zapcore/entry.go

```diff
@@ -164,10 +170,28 @@ const (
 	WriteThenGoexit
 	// WriteThenPanic causes a panic after Write.
 	WriteThenPanic
-	// WriteThenFatal causes a fatal os.Exit after Write.
+	// WriteThenFatal causes an os.Exit(1) after Write.
 	WriteThenFatal
 )
 
+// OnWrite implements the OnWrite method to keep CheckWriteAction compatible
+// with the new CheckWriteHook interface which deprecates CheckWriteAction.
+func (a CheckWriteAction) OnWrite(ce *CheckedEntry, _ []Field) {
+	switch a {
+	case WriteThenGoexit:
+		runtime.Goexit()
+	case WriteThenPanic:
+		panic(ce.Message)
+	case WriteThenFatal:
+		exit.Exit()
+	}
+}
+
+var _ CheckWriteHook = WriteThenNoop
+var _ CheckWriteHook = WriteThenGoexit
+var _ CheckWriteHook = WriteThenPanic
+var _ CheckWriteHook = WriteThenFatal
```

Comment:
The check is type-level, not value-level so we can just verify that CheckWriteAction implements the interface.